### PR TITLE
wire in fuchsiaApp

### LINKED ISSUE #33691: `flutter drive` fails on Fuchsia
Flutter crash report; please file at https://github.com/flutter/flutter/issues.

## command
```
flutter drive test_driver/scroll_perf.dart
```
## exception

NoSuchMethodError: NoSuchMethodError: The getter 'id' was called on null.
Receiver: null
Tried calling: id

```
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:50:5)
#1      FuchsiaDevice.stopApp (package:flutter_tools/src/fuchsia/fuchsia_device.dart:323:67)
<asynchronous suspension>
#2      _stopApp (package:flutter_tools/src/commands/drive.dart:324:45)
<asynchronous suspension>
#3      _startApp (package:flutter_tools/src/commands/drive.dart:240:19)
<asynchronous suspension>
#4      DriveCommand.runCommand (package:flutter_tools/src/commands/drive.dart:118:51)
<asynchronous suspension>
#5      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:488:18)
#6      _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:77:64)
#7      _rootRunUnary (dart:async/zone.dart:1132:38)
#8      _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#9      _FutureListener.handleValue (dart:async/future_impl.dart:126:18)
#10     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:639:45)
#11     Future._propagateToListeners (dart:async/future_impl.dart:668:32)
#12     Future._complete (dart:async/future_impl.dart:473:7)
#13     _SyncCompleter.complete (dart:async/future_impl.dart:51:12)
#14     _AsyncAwaitCompleter.complete.<anonymous closure> (dart:async-patch/async_patch.dart:33:20)
#15     _rootRun (dart:async/zone.dart:1124:13)
#16     _CustomZone.run (dart:async/zone.dart:1021:19)
#17     _CustomZone.bindCallback.<anonymous closure> (dart:async/zone.dart:947:23)
#18     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#19     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
#20     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:115:13)
#21     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:172:5)
```

## flutter doctor

```
[32m[✓][39m Flutter (Channel master, v1.6.8-pre.18, on Linux, locale en_US.UTF-8)
    [32m•[39m Flutter version 1.6.8-pre.18 at /usr/local/google/home/dnfield/src/flutter
    [32m•[39m Framework revision 3ebebebb8d (3 hours ago), 2019-05-31 13:19:44 -0700
    [32m•[39m Engine revision a32df2c928
    [32m•[39m Dart version 2.3.2 (build 2.3.2-dev.0.0 445a23a9bc)

[32m[✓][39m Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    [32m•[39m Android SDK at /usr/local/google/home/dnfield/Android/Sdk
    [32m•[39m Android NDK location not configured (optional; useful for native profiling support)
    [32m•[39m Platform android-28, build-tools 28.0.3
    [32m•[39m ANDROID_HOME = /usr/local/google/home/dnfield/Android/Sdk
    [32m•[39m Java binary at: /opt/android-studio-with-blaze-3.4/jre/bin/java
    [32m•[39m Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b16-5323222)
    [32m•[39m All Android licenses accepted.

[32m[✓][39m Android Studio (version 3.4)
    [32m•[39m Android Studio at /opt/android-studio-with-blaze-3.4
    [32m•[39m Flutter plugin version 23.2.1
    [32m•[39m Dart plugin version 171.4424
    [32m•[39m Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b16-5323222)

[32m[✓][39m VS Code (version 1.34.0)
    [32m•[39m VS Code at /usr/share/code
    [32m•[39m Flutter extension version 3.0.2

[32m[✓][39m Connected device (1 available)
    [32m•[39m crush-ship-baton-easel • 192.168.42.68 • fuchsia • Fuchsia

[32m•[39m No issues found!
```

Potentially related to #33689 

/cc @jonahwilliams @zanderso @digiter

### PR DESCRIPTION
**Pull request by dnfield**
## Description

Fixes #33691 - but does not really add much in the way of actually validating the project (there are other related TODOs for that).

`flutter drive` was failing in this case because the Fuchsia branch of this logic was always returning null.

There are other futher errors related to the timeline data that I'll file a separate issue for.

## Tests

I added the following tests:

Tests for Fuchsia app creation.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Review by zanderso — APPROVED** [review]

File: packages/flutter_tools/test/application_package_test.dart [hunk 1]
```diff
@@ -305,6 +306,49 @@ void main() {
...
     }, overrides: overrides);
   });
+
+  group('FuchsiaApp', () {
+    final Map<Type, Generator> overrides = <Type, Generator>{
+      FileSystem: () => MemoryFileSystem(),
+      Platform: _kNoColorTerminalPlatform,
+      OperatingSystemUtils: () => MockOperatingSystemUtils(),
+    };
+    testUsingContext('Error on non-existing file', () {
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/application_package_test.dart]: nit: spaces between calls to `testUsingContext`

Comment (dnfield) [reply packages/flutter_tools/test/application_package_test.dart]: Done

**Comment by amorenew** [comment]
@dnfield I wonder is Fuchsia an OS to replace Chrome OS or what is Fuchsia app here?

**Comment by dnfield** [comment]
@amorenew - that's not really my area of expertise.  You could read more about Fuchsia in general here: https://fuchsia.googlesource.com/fuchsia/+/master/docs

**Pull request merged**